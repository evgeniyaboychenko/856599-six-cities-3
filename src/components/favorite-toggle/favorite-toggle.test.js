import React from 'react';
import renderer from 'react-test-renderer';
import FavoriteToggle from './favorite-toggle.jsx';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import NameSpace from '../../reducer/name-space.js';

const mockStore = configureStore([]);

const dataComponent = {
  nameClass: `favorite`,
  width: `20`,
  height: `20`
};

it(`should AboutOffer render correctly`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      isFavorite: false,
    }});
  const tree = renderer.create(
      <Provider store={store}>
        <FavoriteToggle
          isFavorite = {true}
          idCard = {1}
          dataComponent = {dataComponent}
          onButtonClick = {()=>{}}
        />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

