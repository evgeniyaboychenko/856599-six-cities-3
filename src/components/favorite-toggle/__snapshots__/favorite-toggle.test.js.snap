// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should AboutOffer render correctly 1`] = `
<button
  className="favorite__bookmark-button place-card__bookmark-button--active button"
  onClick={[Function]}
  type="button"
>
  <svg
    className="place-card__bookmark-icon"
    height="20"
    width="20"
  >
    <use
      href="#icon-bookmark"
    />
  </svg>
  <span
    className="visually-hidden"
  >
    In bookmarks
  </span>
</button>
`;
